packages/react-aria-components/src/Virtualizer.tsx

```diff
@@ -24,73 +24,88 @@ export interface LayoutOptionsDelegate<O> {
 
 interface ILayout<O> extends Layout<Node<unknown>, O>, Partial<DropTargetDelegate>, LayoutOptionsDelegate<O> {}
 
-export interface VirtualizerProps {
+export interface VirtualizerProps<O> {
   /** The child collection to virtualize (e.g. ListBox, GridList, or Table). */
   children: ReactNode,
   /** The layout object that determines the position and size of the visible elements. */
-  layout: ILayout<any>
+  layout: ILayout<O>,
+  /** Options for the layout. */
+  layoutOptions?: O
```

Comment:
Just to make sure I understand this w/ respect to the context causing rerenders in the fake DOM that we discussed previously: this doesn't stop the potential problem where a top level context causes every item in the virtualized collection from re-rendering in the fake DOM right? I'll try this in table after this gets merged